src/scripts/main.js

```diff
@@ -1,3 +1,399 @@
+    </button>
+  </form>
+`);
+
+function callNotification(title, type, text) {
+  const notification = document.createElement('div');
+
+  notification.dataset.qa = 'notification';
+
+  notification.classList.add('notification', type);
+
+  notification.innerHTML = `
+    <h2 class="title">${title}</h2>
+    ${text}
+  `;
+
+  body.append(notification);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 3000);
+};
+
+const form = document.querySelector('form');
+
+form.addEventListener('submit', () => {
+  event.preventDefault();
+
+  const data = new FormData(form);
+
+  const formValues = Object.fromEntries(data.entries());
+
+  const money = +(formValues.salary);
+
+  const newRow = `
+    <tr>
+      <td>
+        ${formValues.name}
+      </td>
+
+      <td>
+        ${formValues.position}
+      </td>
+
+      <td>
+        ${formValues.office}
+      </td>
+
+      <td>
+        ${formValues.age}
+      </td>
+
+      <td>
+        $${money.toLocaleString('en-US')}
+      </td>
+    </tr>
+  `;
+
+  if (formValues.name.length < 4
+    || isNaN(formValues.name) === false
+    || formValues.age < 18
+    || formValues.age > 90
+    || formValues.position.length === 0
+    || isNaN(formValues.position) === false
+    || formValues.salary === 0
+  ) {
+    callNotification('Error', 'error', `
+      <p>You need to fill all fields</p>
+      <p>Length of name must be not less than 4 letters</p>
+      <p>You don't need to use numbers in Name and Position fields</p>
+      <p>Age must be not less than 18 and not more than 90 years old</p>
+    `);
+  } else {
+    callNotification('Success', 'success', `
+      <p>Information added to the table</p>
+    `);
+
+    bodyOfTable.insertAdjacentHTML('beforeend', newRow);
+  }
+});
+
+bodyOfTable.addEventListener('dblclick', () => {
+  const cellInitialValue = event.target.innerHTML;
+  const row = event.target.parentElement;
+  const position = event.target.cellIndex;
+  const currentCell = row.children[position];
+
+  if (position === 2) {
+    event.target.innerHTML = `
+      <select
+        name="office"
+        data-qa="office"
+        class="cell-input"
+      >
+        <option value="Tokyo">Tokyo</option>
+
+        <option value="Singapore">Singapore</option>
+
+        <option value="London">London</option>
+
+        <option value="New York">New York</option>
+
+        <option value="Edinburgh">Edinburgh</option>
+        
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    `;
+  } else {
+    event.target.innerHTML = `
+    <input class="cell-input"></input>
+  `;
+  }
+
+  const input = document.getElementsByClassName('cell-input')[0];
+
+  if (position === 3) {
+    input.type = 'number';
+    input.min = '18';
+    input.max = '99';
+  }
+
+  if (position === 4) {
+    input.type = 'number';
+  }
+
+  input.focus();
+
+  const success = `<p>Information has added to the cell</p>`;
+
+  input.addEventListener('blur', () => {
```

Comment:
move validation into separate function and use it here & in form

Reply:
For the different cell-inputs i use different checks, so i  don't realy know what function i need to create, to use it on blur.
 Maybe you mean that i need to move all checks, that i used in blur event to the separate function and than use it in the event, but what a point, previosly i was move similar function that called "conversion" from the sort event.
 
 P.S.
 Sorry for my bad English, i can explain on russian or ukrainian if you want :)